pkg/controllers/provisioning/provisioner.go

```diff
@@ -212,7 +212,11 @@ func (p *Provisioner) NewScheduler(ctx context.Context, pods []*v1.Pod, stateNod
 		return nil, err
 	}
 	nodePoolList.Items = lo.Filter(nodePoolList.Items, func(n v1beta1.NodePool, _ int) bool {
-		return n.DeletionTimestamp.IsZero()
+		vaildNodePool := n.Spec.Template.Spec.Validate()
```

Comment:
```suggestion
		vaildNodePool := n.Validate()
```

Should this validate function just exist at the top-level?